grails-app/services/com/recomdata/transmart/data/export/HighDimExportService.groovy

```diff
@@ -43,17 +42,20 @@ class HighDimExportService {
      *        factory, e.g., <code>names</code> for the <code>genes</code> filter type. 
      */
     def DataConstraint createFilterConstraints(HighDimensionDataTypeResource dataTypeResource, List<Map> filters) {
-        def dataConstraints = []
+        def dataConstraints = [:]
         filters.each { Map filter ->
-            log.info "creating filter of type ${filter.type}..."
+            log.debug "creating filter of type ${filter.type}..."
             def type = filter.type
             def data = filter.findAll { it.key != 'type' && it.key != 'id' }
-            log.info "  data: ${data}"
-            def constraint = dataTypeResource.createDataConstraint(data, type)
-            log.info "  constraint: ${constraint}"
-            dataConstraints << constraint
+            log.debug "  data: ${data}"
+            if (!dataConstraints.containsKey(type)) {
+                dataConstraints[type] = []
+            }
+            dataConstraints[type] << data
         }
-		DisjunctionDataConstraint disjunction = new DisjunctionDataConstraint(constraints: dataConstraints)
+        DataConstraint disjunction = dataTypeResource.createDataConstraint(
+                ['subconstraints': dataConstraints],
```

Comment:
you don't need to use quotes around key strings in groovy. Instead you need to use parens if you want to use a variable.